### ISSUE TITLE
ContentSwitcher selection with arrow keys

### ISSUE THREAD

**Issue by Zoe-Gathercole1** [carbon-design-system/carbon#6162]
## Environment

> Operating system
MacOS

> Browser
Chrome, FireFox

> Automated testing tool and ruleset

> Assistive technology used to verify

## Detailed description

> What version of the Carbon Design System are you using?
10

> What did you expect to happen?
When using the left/right arrow keys to navigate a ContentSwitcher I expected the correct button to be selected with focus on the desired button. 

> What happened instead?
For example:
I tabbed to the first ContentSwitcher item and pressed the right arrow key. The index passed to the onChange event was the index of the currently selected button so the content did not change as the button to the right was not selected. However, focus still moved to that button.
The issue is the same when using the left arrow key. Every subsequent arrow press means focus is on the wrong button and does not align with the content displayed.

> What WCAG 2.1 checkpoint does the issue violate?

## Steps to reproduce the issue

1. Tab to ContentSwitcher
2. Press either the left or right arrow key

## Additional information

This is what's captured in the onChange function (see _ref) when clicking on the desired button (Sample Code button)
![image](https://user-images.githubusercontent.com/62894097/83254502-c3369a00-a1a6-11ea-9870-08f273fc19c4.png)

This is what's captured in the onChange function (see _ref) when trying to move off the Sample Code button to another on it's left. (pressing the left arrow key on that button)
![image](https://user-images.githubusercontent.com/62894097/83254560-e2352c00-a1a6-11ea-894c-26a33bc05ff1.png)

I would expect the last scenario to say `index: 0` 

Related issue: https://github.ibm.com/mhub/qp-planning/issues/5510

**Comment by emyarod** [comment]
this change appears to be intentional (#2852) but we can reevaluate the differences in behavior between selection via mouse vs selection via arrow keys (both in manual and automatic selection modes), especially since modifying the change event handler would be a breaking change

**Comment by matthew-chirgwin** [comment]
Hi @emyarod - I asked @Zoe-Gathercole1 to raise this issue given an a11y pass of the product we work on where we found this. 

In an ideal world, the on click and on key press interaction would call our onChange with the same result content (ie I click on index 0, I am given 0, user presses arrow key left from index 1, the callback says index is 0 as a result etc). Honestly, if the ContentSwitcher worked like the Tab/Tabs (where we can nest our content in the child, and let the Tab/Tabs deal with which is selected) that would make our lives alot easier!

As is, we as a consumer of the ContentSwitcher need to know some of the internals of how it works to know how to respond (ie if ArrowLeft, -1 from index etc). If this were to change, we would break though. Would be interested to know what you would reccomend we do for now.

**Comment by emyarod** [comment]
> In an ideal world, the on click and on key press interaction would call our onChange with the same result content (ie I click on index 0, I am given 0, user presses arrow key left from index 1, the callback says index is 0 as a result etc).

yeah I believe that is a reasonable expectation as well. I don't currently have a recommendation on what to do at the moment, but just wanted to provide more information about how this pattern came about

I don't fully have the original context around the reason behind the initial change, but if this pattern were to change it would most likely be in a major version release

**Comment by gptt916** [comment]
Any updates on this issue? We are running into the same problem, where the behaviour of clicking on a selection vs using the arrow keys are different due to the `index` value being different.

https://codesandbox.io/s/laughing-gould-ywgsj?file=/src/index.js

I made a quick codesandbox to demonstrate this: When clicking on the buttons, they give the correct index, i.e. clicking on the first button shows `0`, 2nd shows `1` etc. but when navigating using the arrow keys after it shows the previous selection's index.

**Comment by motou** [comment]
It seems that the key event only knows the current section, so it passes the index to the `onChange` event. 

I can take a look at it.

**Comment by joshblack** [comment]
Closed by: https://github.com/carbon-design-system/carbon/pull/6257
